### DISCUSSION TITLE
Create swift App using tracktion engine

### DISCUSSION THREAD

**Discussion by nicolasvair** [Tracktion/tracktion_engine#235] [Q&A]
Hi,

I would like to create the simplest daw possible in swift targeting mac & iPad. 

Anything I should consider, any tips on how to start ? 

My plan would be to use swift to code the UI and tracktion engine for all the audio stuff. 

I suppose Xcode would be the ideal ide for the swift part, but which ide would you recommend for the tracktion side ? Would you rather do this directly in juice editor or consider using chat gpt « multiple files mode », vs code or cursor ?  
i will definitely need ai to help me and pasting to gpt isn’t a great workflow. 

Anyone have experience with those ? 

I’m sorry for being so generic, I’m just not sure where to start. 

Thank you

**Comment by drowaudio** [comment]
There's a few questions that spring to mind:
- Is this a hobby project or a commercial one?
- What do you mean by "simplest DAW possible". Do you have a design or feature list?
- Is this just for macOS?

If you're using Swift for the UI, I'd just use Xcode for the whole IDE. You want to be able to debug all the code properly.

**Comment by nicolasvair** [comment]
Hi Dave,

Thank you for your answer.

I believe we discussed the project this summer by email. 

It wouldn't be a commercial project until having a working beta, which could take some years if I'm alone doing this. 

By "simplest daw possible" I mean having transport, ability to play audio files, basic editing like moving and splitting. It's my first step.
It would be for macOS and iPad. 

The problem I face with using only Xcode is that the workflow with generative AI like copy pasting isn't great atm. So I'm considering using Xcode to run emulators but having another code editor in parallel with AI enabled features (maybe I should have said code editor rather than IDE sorry). 

I know Apple announced basic AI features in last Xcode versions but from my understanding it seems quite basic for now. And I will need a tool that is aware of the other files and context while giving answer, and doing this with a chat interface is really sub-optimal.

**Comment by drowaudio** [comment]
Ah yes I knew I recognised your name. Apologies, things have been really busy at the moment.

Well you can use whatever you want for editing, it just depends on how much semantic information you need. Personally I use CLion but that's for C++ only projects. You could also try VSCode, I think that has good GitHub AI integration. You could edit there and build/run in Xcode.

**Comment by nicolasvair** [comment]
Ok thank you ! I'll try this road.
